iXBRLViewerPlugin/ui.py

```diff
@@ -72,33 +80,96 @@ def __init__(self, cntlr):
         self.grab_set()
         self.wait_window(self)
 
-   
     def close(self, event=None):
+        """
+        Closes the dialog without performing "ok" actions
+        :param event:
+        :return:
+        """
         self.parent.focus_set()
         self.destroy()
 
+    def getTitle(self) -> str:
+        pass
+
+    def addFields(self, frame: Frame, y: int) -> int:
+        """
+        Adds fields specific to the viewer dialog implemented
+        :param frame: Frame to add fields to
+        :param y: The row (y) to start adding fields at
+        :return: The row (y) after adding fields
+        """
+        pass
+
+    def ok(self, event=None):
+        """
+        Actions to perform if OK is pressed
+        :param event:
+        :return:
+        """
+        pass
+
+
+class SaveViewerDialog(ViewerDialog):
+    """
+    Dialog for saving a viewer instance.
+    Initializes with (but doesn't overwrite) default settings configured in the settings dialog.
+    """
+
+    def __init__(self, cntlr):
+
+        self.cntlr = cntlr
+        self.parent = cntlr.parent
+
+        self._filename = StringVar()
+        self._filename.set(self.cntlr.config.setdefault(CONFIG_OUTPUT_FILE, ""))
+        self._zipViewerOutput = BooleanVar()
+        self._zipViewerOutput.set(self.cntlr.config.setdefault(CONFIG_ZIP_OUTPUT, False))
+        self.accepted = False
+
+        super(SaveViewerDialog, self).__init__(cntlr, "Save iXBRL Viewer")
+
+    def addFields(self, frame: Frame, y: int) -> int:
+        y += 1
+        filenameLabel = Label(frame, text="iXBRL file")
+        filenameEntry = Entry(frame, textvariable = self._filename, width=80)
+        filenameBrowse = Button(frame, text=_("Browse..."), command=self.browseForFile)
+        filenameLabel.grid(row=y, column=0, sticky=W, pady=3, padx=3)
+        filenameEntry.grid(row=y, column=1, sticky=EW, pady=3, padx=3)
+        filenameBrowse.grid(row=y, column=2, sticky=EW, pady=3, padx=3)
+        y += 1
+        zipViewerOutputCheckbutton = Checkbutton(frame, text="Zip Viewer Output", variable=self._zipViewerOutput, onvalue=True, offvalue=False)
+        zipViewerOutputCheckbutton.grid(row=y, column=0, pady=3, padx=3)
+        return y
 
     def ok(self, event=None):
-        self.cntlr.config['iXBRLViewerOutputFile'] = self._filename.get()
-        self.cntlr.config['iXBRLViewerScriptURL'] = self._scriptUrl.get()
-        self.cntlr.config["iXBRLViewerFileDir"] = os.path.dirname(self._filename.get())
+        """
+        Saves output-specific config values before triggering instance to be saved.
+        :param event:
+        :return:
+        """
+        # Save output-specific configs for next time
+        self.cntlr.config[CONFIG_FILE_DIRECTORY] = os.path.dirname(self._filename.get())
+        self.cntlr.config[CONFIG_OUTPUT_FILE] = self._filename.get()
+        self.cntlr.config[CONFIG_ZIP_OUTPUT] = self._zipViewerOutput.get()
```

Comment:
An option to somehow reset ixbrl-viewer settings to their defaults would be nice.

Reply:
Added `Reset Defaults` dialog button that resets the dialog variable values, which can then be saved normally by pressing `OK`. I preferred to do it this way rather than resetting, saving, and closing the dialog with one click because it allows the user to view the defaults and cancel or make changes before committing to the defaults.